### PR TITLE
Randomized Quick Sort implementation in C++

### PR DESCRIPTION
**Pull request by SubhradeepSS**
Fixes #1477 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- C++
- 

#### Files Added:
-  [Randomized Quick Sort.cpp](https://github.com/SubhradeepSS/Algo_Ds_Notes/blob/Quick_Sort/Quick_Sort/Randomized%20Quick%20Sort.cpp)
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Quick_Sort/Randomized Quick Sort.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#include<bits/stdc++.h>
+using namespace std;
+
+/* Partitioning function that takes last element as pivot, places 
+  the pivot element at its correct position in sorted 
+  array, and places all smaller (smaller than pivot) 
+  to left of pivot and all greater elements to right 
+  of pivot */
```
Comment (SakshiShreya) [inline Quick_Sort/Randomized Quick Sort.cpp]: Use 4 spaces here for indentation.

File: Quick_Sort/Randomized Quick Sort.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+int partition_index(int array[],int low,int high){
+    int pivot = array[high];
+    int i = low - 1;
+    for(int j = low; j < high; j++){
+        if(array[j] <= pivot){                 // If current element is smaller than or
```
Comment (SakshiShreya) [inline Quick_Sort/Randomized Quick Sort.cpp]: Move the comment to next line or previous line. One line can't contain more than 80 characters including indentation and spaces.

File: Quick_Sort/Randomized Quick Sort.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        if(array[j] <= pivot){                 // If current element is smaller than or  
+            i++;                               // equal to pivot then increment smaller index and swap
+            swap(array[i],array[j]);
+        }
+    }
+    swap(array[i+1],array[high]);
+    return i+1;
+} 
+
+// This function generates a random_index in the range [low,high] and swaps array[high] with array[random_index]
```
Comment (SakshiShreya) [inline Quick_Sort/Randomized Quick Sort.cpp]: Split the comments into multiple lines. Maintain character limit.

File: Quick_Sort/Randomized Quick Sort.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    print_array(array,n);
+    
+    quick_sort(array,0,n-1);
+    cout << "Sorted array: ";
+    print_array(array,n);
+}
+
+/* Sample Input - Output
+Enter the number of elements of array: 8                                                                                      
+
```
Comment (SakshiShreya) [inline Quick_Sort/Randomized Quick Sort.cpp]: Why are there so many spaces in this line?

File: Quick_Sort/Randomized Quick Sort.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+int partition_index(int array[],int low,int high){
+    int pivot = array[high];
+    int i = low - 1;
+    for(int j = low; j < high; j++){
+        if(array[j] <= pivot){                 // If current element is smaller than or  
+            i++;                               // equal to pivot then increment smaller index and swap
+            swap(array[i],array[j]);
+        }
+    }
+    swap(array[i+1],array[high]);
```
Comment (SakshiShreya) [inline Quick_Sort/Randomized Quick Sort.cpp]: Add a space after comma

File: Quick_Sort/Randomized Quick Sort.cpp [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+array[] - Array to be sorted, 
+low - Starting index, 
+high - Ending index */
+void quick_sort(int array[],int low,int high){
+    if(low < high){
+        int partition = random_partition(array,low,high);
+        // Separately sort elements before 
+        // partition and after partition 
+        quick_sort(array,low,partition-1);
+        quick_sort(array,partition+1,high);
```
Comment (SakshiShreya) [inline Quick_Sort/Randomized Quick Sort.cpp]: Add space around operators.

**Comment by SubhradeepSS** [comment]
@SakshiShreya why did you close the PR?

**Comment by SubhradeepSS** [comment]
@jainaman224 Please open this PR

**Comment by SubhradeepSS** [comment]
@jainaman224 @SakshiShreya I have made the changes as requested but my PR is still closed.Please open and review my PR

**Comment by SubhradeepSS** [comment]
@jainaman224 @somya-kapoor Please open and review this PR also.

**Comment by SubhradeepSS** [comment]
@jainaman224 @somya-kapoor Please review this PR

**Comment by SubhradeepSS** [comment]
> 
> 
> @jainaman224 @somya-kapoor Please review this PR

The issue #1478 was assigned me a lot ago on 2nd March and I completed the task on 2nd march itself but the PR was closed.Please review this PR also.

**Comment by SubhradeepSS** [comment]
@jainaman224 @somya-kapoor Please review the PR for issue #1478

**Comment by somya-kapoor** [comment]
@SubhradeepSS It was closed due to plagiarism.

**Pull request closed without merging**